Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,286 @@
+                if (rollsRemaining == 3)
+                {
+                    roll3 = bowlingMarks.Substring(placeCount + 2, 1);  // pointer for roll 3
+
+                    switch (roll3)
+                    {
+                        case ("-"):  // OPEN frame no points
+                            frame.bonusRoll = 0;
+                            frame.frameOutcome = "OPEN";
+                            break;
+
+                        case ("X"):  // STRIKE
+                            frame.bonusRoll = 10;
+                            frame.frameOutcome = "STRIKE";
+                            break;
+
+                        default:  // OPEN frame
+                            frame.bonusRoll = Convert.ToInt32(roll3);
+                            frame.frameOutcome = "OPEN";
+                            break;
+                    }
+                }
+
+                //  calculation of frame 10 rolls
+                frame.frameScore = frame.firstRoll + frame.secondRoll + frame.bonusRoll;
+
+                // add finished frame to the collection
+                framesCollection.Add(frame);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(String.Concat("There was a problem creating the Frame 10 collection: ", ex.Message));
+            }
+        }
+
+        /** This finds any frames where STRIKES or SPARES occurred and updates the frameScore variable based on bowling calculation rules **/
+        public List<Frames> UpdateFrameScores(List<Frames> framesCollection)
+        {
+            /** For STRIKES we would need to get scores from the next two frames' rolls or just the next two rolls to calculate the score.
+                For SPARES we need to know the scores for the next frame's roll.  **/
+
+            // Frame initialization
+            Frames currentFrame;
+            Frames oneFrame;
+            Frames twoFrame;
+
+            /* Iterate through each frame in the collection: frameNumberTotal is adjusted by two to prevent 
+               assigning current, one, and two frames outside of the total 10 objects in collection */
+            for (int i = 0; i <= frameNumberTotal - 2; i++)
+            {
+                currentFrame = framesCollection[i];
+                oneFrame = framesCollection[i + 1];
+
+                // make sure we don't assingn the oneFrame or twoFrame variables to a roll outside of our pointer index (frameNumberTotal)
+                if ((frameNumberTotal - 2) - i >= 2)
+                {
+                    twoFrame = framesCollection[i + 2];
+                }
+                else  // last roll in frame by default
+                {
+                    twoFrame = framesCollection[frameNumberTotal - 1];
+                }
+
+
+                // Adjust frame scores in all frames where a STRIKE or SPARE were detected
+                switch (currentFrame.frameOutcome)
+                {
+                    case ("STRIKE"):
```

Comment:
Using strings like this can be fragile because it's easy to make a typo and the compiler won't know any better.  The code will compile but the code won't work and it could be tricky to debug...I suggest using an enum instead and let the compiler do the heavy work for us!

For instance:

```
switch (currentFrame.frameOutcome)
{
    case("STRIEK"):
        ...
}

I would investigate which other hard-coded strings this could be applicable to.